src/main/java/org/jboss/tools/intellij/openshift/utils/odo/ComponentFeature.java

```diff
@@ -39,32 +41,40 @@ public ComponentFeature getPeer() {
 
     private final List<String> stopArgs;
 
-    ComponentFeature(Kind kind, String label, String output, List<String> startArgs, List<String> stopArgs) {
-        this.kind = kind;
+    ComponentFeature(Mode mode, String label, String output, List<String> startArgs, List<String> stopArgs) {
+        this.mode = mode;
         this.label = label;
         this.output = output;
         this.startArgs = startArgs;
         this.stopArgs = stopArgs;
     }
 
-    ComponentFeature(Kind kind, String label, String output, List<String> startArgs) {
-        this(kind, label, output, startArgs, Collections.emptyList());
+    ComponentFeature(Mode mode, String label, String output, List<String> startArgs) {
+        this(mode, label, output, startArgs, Collections.emptyList());
     }
 
-    public ComponentFeature getPeer() {
-        return null;
-    }
-
-    public Kind getKind() {
-        return kind;
+    public Mode getMode() {
+        return mode;
     }
 
     public String getLabel() {
         return label;
     }
 
+    public List<String> getExtraArgs() {
+        return Collections.emptyList();
+    }
+
     public List<String> getStartArgs() {
-        return startArgs;
+        List<String> result = new ArrayList<>(startArgs);
+        if (!getExtraArgs().isEmpty()) {
```

Comment:
not sure this check is even needed. If you add an empty list, nothing gets added.